DQMOffline/Trigger/plugins/BPHMonitor.cc

```diff
@@ -0,0 +1,263 @@
+      double z0 = track->dz(beamSpot->position());
+      mud0_.denominator ->Fill(d0);
+      muz0_.denominator ->Fill(z0);
+    }
+  } 
+  // applying selection for numerator
+  if (num_genTriggerEventFlag_->on() && ! num_genTriggerEventFlag_->accept( iEvent, iSetup) ) return;
+  for (auto const & m : *muoHandle) {
+    if(!muoSelection_(m))continue;
+    muPhi_.numerator->Fill(m.phi());
+    muEta_.numerator->Fill(m.eta());
+    muPt_.numerator ->Fill(m.pt());
+    const reco::Track * track = 0;
+    if (m.isTrackerMuon()) track = & * m.innerTrack();
+    else if (m.isStandAloneMuon()) track = & * m.outerTrack();
+    if (track) {
+      double d0 = track->dxy(beamSpot->position());
+      double z0 = track->dz(beamSpot->position());
+      mud0_.numerator ->Fill(d0);
+      muz0_.numerator ->Fill(z0);
+    }
+  }
+	
+  // filling histograms (num_genTriggerEventFlag_)  
+
+}
+
+void BPHMonitor::fillHistoPSetDescription(edm::ParameterSetDescription & pset)
+{
+  pset.add<int>   ( "nbins");
+  pset.add<double>( "xmin" );
+  pset.add<double>( "xmax" );
+}
+
+void BPHMonitor::fillHistoLSPSetDescription(edm::ParameterSetDescription & pset)
+{
+  pset.add<int>   ( "nbins", 2500);
+}
+
+void BPHMonitor::fillDescriptions(edm::ConfigurationDescriptions & descriptions)
+{
+  edm::ParameterSetDescription desc;
+  desc.add<std::string>  ( "FolderName", "HLT/BPH/" );
+
+  desc.add<edm::InputTag>( "tracks",  edm::InputTag("generalTracks") );
+  desc.add<edm::InputTag>( "offlinePVs",     edm::InputTag("offlinePrimaryVertices") );
+  desc.add<edm::InputTag>( "beamSpot",edm::InputTag("offlineBeamSpot") );
+  desc.add<edm::InputTag>( "muons",    edm::InputTag("muons") );
+  desc.add<std::string>("muoSelection", "abs(eta)<1.4 & isPFMuon & isGlobalMuon  & innerTrack.hitPattern.trackerLayersWithMeasurement>5 & innerTrack.hitPattern.numberOfValidPixelHits>0");
+  desc.add<std::string>("muoSelection_ref", "isPFMuon & isGlobalMuon  & innerTrack.hitPattern.trackerLayersWithMeasurement>5 & innerTrack.hitPattern.numberOfValidPixelHits>0");
+  desc.add<int>("nmuons",     1);
+
+  edm::ParameterSetDescription genericTriggerEventPSet;
+  genericTriggerEventPSet.add<bool>("andOr");
+  genericTriggerEventPSet.add<edm::InputTag>("dcsInputTag", edm::InputTag("scalersRawToDigi") );
+  genericTriggerEventPSet.add<std::vector<int> >("dcsPartitions",{});
+  genericTriggerEventPSet.add<bool>("andOrDcs", false);
+  genericTriggerEventPSet.add<bool>("errorReplyDcs", true);
+  genericTriggerEventPSet.add<std::string>("dbLabel","");
+  genericTriggerEventPSet.add<bool>("andOrHlt", true);
+  genericTriggerEventPSet.add<edm::InputTag>("hltInputTag", edm::InputTag("TriggerResults::HLT") );
+  genericTriggerEventPSet.add<std::vector<std::string> >("hltPaths",{});
+  genericTriggerEventPSet.add<std::string>("hltDBKey","");
+  genericTriggerEventPSet.add<bool>("errorReplyHlt",false);
+  genericTriggerEventPSet.add<unsigned int>("verbosityLevel",1);
```

Comment:
Hi @anigamova - a late followup on this pull request - for production running this parameter needs to be 0. Please make a new PR with that change. Thanks